pkg/liquidity-mining/test/GaugeMaxRelativeWeight.test.ts

```diff
@@ -0,0 +1,213 @@
+
+    await gaugeController.add_type('Ethereum', 0);
+  });
+
+  sharedBeforeEach('deploy token mocks', async () => {
+    token0 = await deploy('TestBalancerToken', { args: [admin.address, 'Balancer-LP0', 'LP0'] });
+    BAL = await deploy('TestBalancerToken', { args: [admin.address, 'Balancer', 'BAL'] });
+  });
+
+  sharedBeforeEach('deploy gauge implementation and factory', async () => {
+    const balTokenAdmin = await deploy('MockBalancerTokenAdmin', { args: [vault.address, BAL.address] });
+    const balMinter = await deploy('BalancerMinter', { args: [balTokenAdmin.address, gaugeController.address] });
+
+    // We won't be using the code that requires the VotingEscrowDelegationProxy, so we just use any address.
+    liquidityGaugeImplementation = await deploy('LiquidityGaugeV5', {
+      args: [balMinter.address, ANY_ADDRESS, adaptor.address],
+    });
+    liquidityGaugeFactory = await deploy('LiquidityGaugeFactory', { args: [liquidityGaugeImplementation.address] });
+    stakelessGaugeFactory = await deploy('SingleRecipientGaugeFactory', { args: [balMinter.address] });
+  });
+
+  sharedBeforeEach('deploy gauges', async () => {
+    liquidityGauge = await deployedAt(
+      'LiquidityGaugeV5',
+      await deployGauge(liquidityGaugeFactory, token0.address, 'GaugeCreated')
+    );
+    stakelessGauge = await deployedAt(
+      'SingleRecipientGauge',
+      await deployGauge(stakelessGaugeFactory, token0.address, 'SingleRecipientGaugeCreated')
+    );
+  });
+
+  sharedBeforeEach('set up permissions', async () => {
+    const action = await actionId(adaptor, 'set_max_relative_weight', liquidityGaugeImplementation.interface);
+    await vault.grantPermissionsGlobally([action], admin);
+  });
+
+  async function deployGauge(gaugeFactory: Contract, poolAddress: string, eventName: string): Promise<string> {
+    const tx = await gaugeFactory.create(poolAddress);
+    const event = expectEvent.inReceipt(await tx.wait(), eventName);
+
+    return event.args.gauge;
+  }
+
+  async function setNewMaxRelativeWeight(maxRelativeWeight: BigNumber) {
+    const calldata = gauge.interface.encodeFunctionData('set_max_relative_weight', [maxRelativeWeight]);
+    return adaptor.connect(admin).performAction(gauge.address, calldata);
+  }
+
+  describe('LiquidityGaugeV5', () => {
+    beforeEach('use liquidity gauge', () => {
+      gauge = liquidityGauge;
+    });
+    itTestsMaxRelativeWeightForGauge();
+  });
+
+  describe('StakelessGauge', () => {
+    beforeEach('use stakeless gauge', () => {
+      gauge = stakelessGauge;
+    });
+    itTestsMaxRelativeWeightForGauge();
+  });
+
+  function itTestsMaxRelativeWeightForGauge() {
+    describe('set_maximum_relative_weight', () => {
+      const defaultMaxRelativeWeight = fp(1.0);
+      const newMaxRelativeWeight = fp(0.3);
+      const aboveAbsoluteMaxRelativeWeight = defaultMaxRelativeWeight.add(1);
+
+      context('without permissions', () => {
```

Comment:
```suggestion
      context('when the caller is not authorized', () => {
```
It's better to be a bit more explicit about these things, even if it results in added verbosity